service/k8sclient/internal/kube/kube.go

```diff
@@ -436,3 +765,23 @@ func deleteObject(helper *resource.Helper, namespace, name string) error {
 	}
 	return nil
 }
+
+// translateMicroTimestampSince returns the elapsed time since timestamp in
+// human-readable approximation.
+func translateMicroTimestampSince(timestamp metav1.MicroTime) string {
+	if timestamp.IsZero() {
+		return "<unknown>"
+	}
+
+	return duration.HumanDuration(time.Since(timestamp.Time))
+}
+
+// translateTimestampSince returns the elapsed time since timestamp in
+// human-readable approximation.
+func translateTimestampSince(timestamp metav1.Time) string {
+	if timestamp.IsZero() {
+		return "<unknown>"
+	}
+
+	return duration.HumanDuration(time.Since(timestamp.Time))
+}
```

Comment:
Seems like a good case for a generic function.

Reply:
I'm not sure that we need to spend time writing a generic function for that case because the only place where it's used is to get events for statefulset. Also, it was copied from kubectl code to get events to get rid of kubectl usage for 'Logs' feature in the PMM/DBaaS UI. However, It looks like we won't need to call
```
kubectl get logs
kubectl get events
```
to understand what's going wrong with the cluster because we maybe adopt Loki and will have the integration with Loki for logs. Also, I assume that in the nearest future It'll be redesigned to make the user experience more smooth because right now it's hard to understand what's going on with the cluster in the UI.

Reply:
Sure, this isn't super relevant and one could even argue that having the two functions is better to have a separation of concerns just like `kubernetes/apimachinery` found the need to have the two types. I only thought about generics because the logic is exactly the same on both but feel free to ignore this comment.